### ISSUE TITLE
Why the color loss can optimize the generator?

### ISSUE THREAD

**Issue by jj199603** [sharif-apu/BJDD_CVPR21#11]
Dear author, I'm confused about the function of the color loss function. I find you detach the generated tensor from the foward graph and convert it to numpy array. However, after these operations, the grads cannot backpropagate to the network. So why the color loss can optimize the parameters of the network???

**Comment by tonyzzzt** [comment]
The same problem.
Have you found a reasonable explanation? @jj199603 @sharif-apu
